### PR TITLE
feat(filter-between-dates): filter AGO items by modified date

### PR DESCRIPTION
**Pull request by pranavkulkarni**
filter AGO items by modified date

AFFECTS PACKAGES:
@esri/hub-common
@esri/hub-search

### CODE REVIEW

**Comment by pranavkulkarni** [comment]
> I am not very familiar with this codebase. But I have two questions:
> 
> 1. AGO range queries can be used on non-date fields. Does this preclude allowing for that in the future?
No. It is flexible for any range query. We are following what v3 api does like `/api/v3/search?filter[modified]=between(2019/10/30,2019/10/31)`

> 2. Wouldn't it be better to accept dates as a timestamp or a date object rather than a string?
Hub.js is used to serialize query parameters in the UI because of unified search. The same query params are used to search against v3 backend and against AGO using this hub-search package. The serialized query params appear in the URL so I think we decided to go with dates like `2019/10/31` rather than unix timestamp. The v3 backend converts that to unix timestamp automatically thanks to Elasticsearch. We need the same thing to happen in hub.js which is used by the backend to query AGO.

File: packages/common/src/util.ts [hunk 1]
```diff
@@ -309,3 +309,19 @@ export function extend(
...
+/**
+ * Add number of days to a given date
+ *
+ * @export
+ * @param {string} date
+ * @param {number} numOfDays
+ * @returns {string} date string with numOfDays added to the given date
+ */
+export function addDays(date: string, numOfDays: number): string {
+  const given = new Date(date);
```
Comment (trescube) [inline packages/common/src/util.ts]: Is there error handling in place to cover the case where `new Date(date)` returns `Invalid Date`?

Comment (pranavkulkarni) [reply packages/common/src/util.ts]: Yes, It is handled by hub-indexer here - https://github.com/ArcGIS/hub-indexer/blob/master/packages/api/search/items.js#L27-L32

Comment (trescube) [reply packages/common/src/util.ts]: It's not going to get that far:  

```
> addDays('this is not a date', 1)
RangeError: Invalid time value
    at Date.toISOString (<anonymous>)
    at addDays (repl:5:5)
```

**Pull request merged**